pkg/sdk/schemas.go

```diff
@@ -13,3 +52,335 @@ func (v *Schema) ID() SchemaIdentifier {
+type DropSchemaOptions struct {
+	drop     bool             `ddl:"static" sql:"DROP"` //lint:ignore U1000 This is used in the ddl tag
+	schema   bool             `ddl:"static" sql:"SCHEMA"`
+	IfExists *bool            `ddl:"keyword" sql:"IF EXISTS"`
+	name     SchemaIdentifier `ddl:"identifier"`
+	// one of
+	Cascade  *bool `ddl:"static" sql:"CASCADE"`
+	Restrict *bool `ddl:"static" sql:"RESTRICT"`
+}
+
+func (opts *DropSchemaOptions) validate() error {
+	var errs []error
+	if !validObjectidentifier(opts.name) {
+		errs = append(errs, ErrInvalidObjectIdentifier)
+	}
+	if everyValueSet(opts.Cascade, opts.Restrict) {
+		errs = append(errs, errors.New("only one of the fields [ Cascade | Restrict ] can be set at once"))
+	}
+	return errors.Join(errs...)
+}
+
+func (v *schemas) Drop(ctx context.Context, id SchemaIdentifier, opts *DropSchemaOptions) error {
+	if opts == nil {
+		opts = &DropSchemaOptions{}
+	}
+	opts.name = id
+	if err := opts.validate(); err != nil {
+		return err
+	}
+	sql, err := structToSQL(opts)
+	if err != nil {
+		return err
+	}
+	_, err = v.client.exec(ctx, sql)
+	return err
+}
+
+type undropSchemaOptions struct {
+	undrop bool             `ddl:"static" sql:"UNDROP"` //lint:ignore U1000 This is used in the ddl tag
+	schema bool             `ddl:"static" sql:"SCHEMA"` //lint:ignore U1000 This is used in the ddl tag
+	name   SchemaIdentifier `ddl:"identifier"`
+}
+
+func (opts *undropSchemaOptions) validate() error {
+	if !validObjectidentifier(opts.name) {
+		return ErrInvalidObjectIdentifier
+	}
+	return nil
+}
+
+func (v *schemas) Undrop(ctx context.Context, id SchemaIdentifier) error {
+	opts := &undropSchemaOptions{
+		name: id,
+	}
+	if err := opts.validate(); err != nil {
+		return err
+	}
+	sql, err := structToSQL(opts)
+	if err != nil {
+		return err
+	}
+	_, err = v.client.exec(ctx, sql)
+	return err
+}
+
+type describeSchemaOptions struct {
+	describe bool             `ddl:"static" sql:"DESCRIBE"` //lint:ignore U1000 This is used in the ddl tag
+	database bool             `ddl:"static" sql:"SCHEMA"`   //lint:ignore U1000 This is used in the ddl tag
+	name     SchemaIdentifier `ddl:"identifier"`            //lint:ignore U1000 This is used in the ddl tag
+}
+
+func (opts *describeSchemaOptions) validate() error {
+	if !validObjectidentifier(opts.name) {
+		return ErrInvalidObjectIdentifier
+	}
+	return nil
+}
+
+type SchemaDetails struct {
+	CreatedOn time.Time `db:"created_on"`
+	Name      string    `db:"name"`
+	Kind      string    `db:"kind"`
+}
+
+func (v *schemas) Describe(ctx context.Context, id SchemaIdentifier) ([]SchemaDetails, error) {
+	opts := &describeSchemaOptions{
+		name: id,
+	}
+	if err := opts.validate(); err != nil {
+		return nil, err
+	}
+	sql, err := structToSQL(opts)
+	if err != nil {
+		return nil, err
+	}
+	var details []SchemaDetails
+	err = v.client.query(ctx, &details, sql)
+	if err != nil {
+		return nil, err
+	}
+	return details, err
+}
+
+type InSchema struct {
```

Comment:
this name is a bit confusing. im not sure what our naming convention is, but "InSchema" sounds like something that would be in a schema. As in a schema object. Whereas this is more about for showing schemas in something (database or account). perhaps "SchemaIn" would be more appropriate

Reply:
Right, changed to SchemaIn